docs/pages/api-docs/data-grid/data-grid.md

```diff
@@ -84,6 +89,7 @@ import { DataGrid } from '@material-ui/data-grid';
 | <span class="prop-name">onRowLeave</span> | <span class="prop-type">(param: GridRowParams, event: React.MouseEvent) => void</span> |   | Callback fired when a mouse leave event comes from a row container element. |
 | <span class="prop-name">onSelectionModelChange</span> | <span class="prop-type">(param: GridSelectionModelChangeParams) => void</span> |   | Callback fired when the selection state of one or multiple rows changes. |
 | <span class="prop-name">onSortModelChange</span> | <span class="prop-type">(param: GridSortModelParams) => void</span> |   | Callback fired when the sort model changes before a column is sorted. |
+| <span class="prop-name">onStateChange</span> | <span class="prop-type">(params: any) => void</span> |   | Callback fired when the state of the grid is updated. |
```

Comment:
Private API

Reply:
Well, if it's private we should consider removing this prop. It's no used internally.

Reply:
I see it being used here https://github.com/mui-org/material-ui-x/blob/e6bed6b36d362810e49af90e4c6271d3c41ba93c/packages/grid/_modules_/grid/hooks/features/core/useGridApi.ts#L38
Am I missing something?

Reply:
I was talking about the prop. We don't use it internally. XGrid's users could still get state updates by subscribing to `GRID_STATE_CHANGE`.

Reply:
@m4theushw We have mentioned it in some of the issues. I think that we should consider it experimental at this point. It's unclear what problem it solves.

Reply:
So based on this conversation there are 2 things:
1. Remove the doc and the prop both.
2. Only remove the doc.

What do we do?

Reply:
In any case, I removed this API from docs.